src/coreclr/nativeaot/Runtime/arm/GcProbe.S

```diff
@@ -7,14 +7,23 @@
     .global RhpGcPoll2
 
     LEAF_ENTRY RhpGcPoll
-        PREPARE_EXTERNAL_VAR_INDIRECT_W RhpTrapThreads, 0
-        cbnz    w0, RhpGcPollRare // TrapThreadsFlags_None = 0
-        ret
+        // @todo: I'm assuming it's not OK to trash any register here. If that's not true we can optimize the
```

Comment:
I do not think that this assumption is correct. It should be fine to trash any volatile registers here.

Reply:
Should I fix that in windows asm too?

Reply:
Yes, it would be nice.